### PR TITLE
Allow the target namespace to vary in testing

### PR DESCRIPTION
**Pull request by marun**
Targets #197

### CODE REVIEW

**Comment by marun** [comment]
Rebased

**Comment by pmorie** [comment]
@marun travis failure seems legit

```
# github.com/kubernetes-sigs/federation-v2/test/integration
test/integration/crud_test.go:67:47: not enough arguments in call to framework.NewSyncControllerFixture
	have ("github.com/kubernetes-sigs/federation-v2/test/common".TestLogger, typeconfig.Interface, *rest.Config, string, string)
	want ("github.com/kubernetes-sigs/federation-v2/test/common".TestLogger, typeconfig.Interface, *rest.Config, string, string, string)
test/integration/integration_test.go:61:60: not enough arguments in call to framework.NewSyncControllerFixture
	have ("github.com/kubernetes-sigs/federation-v2/test/common".TestLogger, typeconfig.Interface, *rest.Config, string, string)
	want ("github.com/kubernetes-sigs/federation-v2/test/common".TestLogger, typeconfig.Interface, *rest.Config, string, string, string)
test/integration/replicaschedulingpreference_test.go:113:46: not enough arguments in call to framework.NewRSPControllerFixture
	have ("github.com/kubernetes-sigs/federation-v2/test/common".TestLogger, *rest.Config, string, string)
	want ("github.com/kubernetes-sigs/federation-v2/test/common".TestLogger, *rest.Config, string, string, string)
test/integration/servicedns_test.go:155:35: not enough arguments in call to servicedns.StartController
	have (*rest.Config, string, string, chan struct {}, bool)
	want (*rest.Config, string, string, string, <-chan struct {}, bool)
FAIL	github.com/kubernetes-sigs/federation-v2/test/integration [build failed]
```

**Comment by marun** [comment]
@pmorie Fixed.

File: test/e2e/framework/managed.go [hunk 1]
```diff
@@ -120,12 +120,14 @@ func (f *ManagedFramework) TestNamespaceName() string {
 func (f *ManagedFramework) SetUpControllerFixture(typeConfig typeconfig.Interface) {
 	// TODO(marun) check TestContext.InMemoryControllers before setting up controller fixture
 	kubeConfig := fedFixture.KubeApi.NewConfig(f.logger)
-	fixture := framework.NewSyncControllerFixture(f.logger, typeConfig, kubeConfig, fedFixture.SystemNamespace, fedFixture.SystemNamespace)
+	ns := fedFixture.SystemNamespace
```
Comment (font) [inline test/e2e/framework/managed.go]: How come we're specifying the `SystemNamespace` here and in `SetUpServiceDNSControllerFixture` below for managed E2E, but in `initCrudTest` we use `metav1.NamespaceAll` for integration?

**Comment by marun** [comment]
@font Good catch, fixed.  CI isn't running managed e2e so the failure wasn't being detected.  That's a point for reconciling managed and integration sometime soon.

**Pull request merged**